snuba/datasets/entities/metrics.py

```diff
@@ -77,16 +77,27 @@ def transform_expression(exp: Expression) -> Expression:
 class MetricsEntity(Entity, ABC):
     def __init__(
         self,
-        writable_storage_key: StorageKey,
+        writable_storage_key: Optional[StorageKey],
         readable_storage_key: StorageKey,
         value_schema: Sequence[Column[SchemaModifiers]],
         mappers: TranslationMappers,
+        allow_cross_org: bool = False,
```

Comment:
Not sure if we prefer to handle this via a param or possibly split this class up into a more generic base class.

Also could probably do with a better name here, but I couldn't think of one.

Reply:
I wonder if it makes sense to just elevate `validators` to a param of this method given that we're already calling `super.__init__()` everywhere -- it could have a default of the current values. It looks like it's a parameter on the base `Entity` class so the pattern already exists

Reply:
Agreed, this is much cleaner